File: Lab5_Csharp/Fabrika.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Linq.Expressions;
using System.Text;
using System.Threading.Tasks;

namespace _17150_Djordje_Jovanovic_PJ_Lab_5
{
    class Fabrika
    {
		private Kolekcija<Traka> k;


		public Fabrika(int n)
		{
			k = new Kolekcija<Traka>(n);
		}

		public void Dodaj(Traka t)
		{
			k.Dodaj(t);
		}

		public void Pokreni(int i, int kol)
		{
			k.Pokreni(i, kol);
		}

		public void Kraj()
		{
			k.Kraj();
		}

		public void Kraj(int i)
		{
			k.Kraj(i);
		}

		public void Ucitaj(string s)
		{
			try
			{
				using (BinaryReader br = new BinaryReader(new FileStream(s, FileMode.Open)))
					k.Ucitaj(br);
			}
			catch(IOException e)
			{
				Console.WriteLine(e.Message);
			}
		}
		public void Pokreni(string s)
		{
			try
			{
				using (BinaryReader br = new BinaryReader(new FileStream(s, FileMode.Open)))
					k.Pokreni(br);
			}
			catch(IOException e)
			{
				Console.WriteLine(e.Message);
			}
		}
	}
}

File: Lab2_java/Fabrika.java
public class Fabrika {
	
	private int maxTraka;
	private int trenTraka;
	private Kolekcija<Traka> KOL;
	
	Fabrika(int k) {
		trenTraka = 0;
		maxTraka = k;
		KOL = new Kolekcija<Traka>(k);
		KOL.tr=new Traka[k];
	}
	
	public void dodaj(Traka t)	{
		if(trenTraka < maxTraka)
			KOL.dodaj(t);
		else
			System.out.println("Nemoguce je dodati vise traka!");
	}
	
	public void pokreni(int i, int kol) {
		KOL.tr[i].pocniSRadom(kol);
	}
	
	public void kraj()	{
		for(int i=0; i<trenTraka; i++)
			KOL.tr[i].staniSRadom();
	}
	
	public void kraj(int i)	{
		KOL.tr[i-1].staniSRadom();
	}
}